src/stan/math/rev/mat/functor/finite_diff_grad_hessian.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef STAN__MATH__REV__MAT__FUNCTOR__FINITE_DIFF_GRAD_HESSIAN_HPP
+#define STAN__MATH__REV__MAT__FUNCTOR__FINITE_DIFF_GRAD_HESSIAN_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/mix/mat/functor/gradient.hpp>
+
+namespace stan {
+  
+  namespace agrad {
+
+    /** 
+     * Calculate the value and the gradient of the hessian of the specified
+     * function at the specified argument using second-order autodiff and
+       * first-order finite difference.  
+     *
+     * <p>The functor must implement 
```

Comment:
What do you mean by a reference?  As in to a book or paper?
I don't think that's necessary.  A definition of how it works
would be better.  

- Bob

> On Mar 17, 2015, at 5:34 AM, Daniel Lee <notifications@github.com> wrote:
> 
> In src/stan/math/rev/mat/functor/finite_diff_grad_hessian.hpp:
> 
> > +#define STAN__MATH__REV__MAT__FUNCTOR__FINITE_DIFF_GRAD_HESSIAN_HPP
> > +
> > +#include <stan/math/prim/mat/fun/Eigen.hpp>
> > +#include <stan/math/rev/core.hpp>
> > +#include <stan/math/mix/mat/functor/gradient.hpp>
> > +
> > +namespace stan {
> > +  
> > +  namespace agrad {
> > +
> > +    /** 
> > +     * Calculate the value and the gradient of the hessian of the specified
> > +     * function at the specified argument using second-order autodiff and
> > +       * first-order finite difference.  
> > +     *
> > +     * <p>The functor must implement 
> 
> include reference for the algorithm. Also indicate number of calls to f
> 
> —
> Reply to this email directly or view it on GitHub.
>